src/briefcase/integrations/linuxdeploy.py

```diff
@@ -47,16 +48,23 @@ def exists(self):
     def install(self):
         """Download and install linuxdeploy or plugin."""
         try:
-            download_path = self.command.download_url(
+            self.command.download_url(
                 url=self.download_url,
                 download_path=self.file_path,
             )
         except requests_exceptions.ConnectionError as e:
             raise NetworkFailure(f"download {self.full_name}") from e
 
-        with self.command.input.wait_bar(f"Installing {self.full_name}..."):
-            self.command.os.chmod(download_path, 0o755)
-            if self.file_name.endswith("AppImage"):
+    def post_install_processing(self):
```

Comment:
This is the only change that doesn't make sense to me - why break this out as a different method? Or, if it's broken out - why not call it as part of `install()`? 

Is there any condition where we'd *not* do post_install is there any condition in which we would install, but *not* post install, or *only* post install without an install occurring? Do you envisage a subclass needing to override/augment post-installation procedures? Unless either of these things are true, I'm not sure I see the value in splitting this method out.

Reply:
So, I struggled a bit with how to handle this. The problem as I perceive it is all current and future install code paths should all run the logic in `post_install_processing()`....so, then, who should be responsible for ensuring it's called?

As with the other tools, the code flow for `verify` and `upgrade` is controlled by the base class; these are important because they are the only way the tool ever gets installed. Since `install()` is being overridden for local plugin files, should that subclass (and future ones like it) take on the responsibility of calling `post_install_processing()` themselves?

I think I see either implementation as mostly equal....have the base class always call `post_install_processing()` or require the subclasses to call it themselves when they override `install()`.

That's my logic, though, if it makes sense. Definitely open to feedback about it.

Reply:
I think I can see where you're coming from; however, from my perspective, the place where that approach falls down is that `uninstall()` and `install()` are atomic operations; but there's no use case (that I can think of) where you'd want `post_install_processing()` to be similarly atomic. Post install processing is something that has to be done for an install to be complete; by splitting it out, you've created a case where someone could do an install, forget that post-install processing is required, and end up with a state where you effectively have an "incomplete" installation. Unless there's a situation where that "incomplete" state is useful, it just seems like a foot-gun waiting to go off.

One approach would be to use the refactoring you've done here and make `post_install_processing()` the last thing that `install()` does, which provides a clean extension point if a different plugin requires different post-install processing - but we don't actually have that requirement, and I can't see a case where it would be required in future, either. 

Given that it's easy enough to add back in if it turns out we need it later, my inclination is to collapse this back in to `install()` and keep the implementation simple - and if it turns out we *do* need this feature in future, you have my permission to keep the receipts so you can say "I told you so" :-)

Reply:
That makes sense....except for the exact implementation....I feel like I might be missing something.

If `LinuxDeployBase.post_install_processing()` doesn't remain its own method and is incorporated back in to `LinuxDeployBase.install()`, wouldn't it need to be effectively duplicated in to `LinuxDeployLocalFilePlugin.install()`?

Reply:
Oh yeah - that little detail... the point of the whole PR ;-) 

In that case, factoring into a method makes perfect sense - but I'd argue it should be invoked from inside `install()`, rather than `verify()`, for the same reason - there's no case (I can see) where you'd want to install, but not have it fully post-installed.